src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenConditionalOperatorTests.cs

```diff
@@ -2763,5 +2763,283 @@ .maxstack  1
   IL_0064:  ret
 }");
         }
+
+        [Fact]
+        [WorkItem(3519, "https://github.com/dotnet/roslyn/issues/35319")]
+        public void ConditionalAccessUnconstrainedTField()
+        {
+            var source = @"using System;
+public class Foo<T>
+{
+    public Foo(T t) => this.t = t;
+    public Foo(){}
+
+    private T t;
+    
+    public void Print()
+    {
+        Console.WriteLine(t?.ToString());
+        Console.WriteLine(t);
+    }
+    
+}
+
+public struct S
+{
+    int a;
+    public override string ToString() => a++.ToString();
+}
+
+public static class Program
+{
+    public static void Main()
+    {
+        new Foo<S>().Print();
+        new Foo<S?>().Print();
+        new Foo<S?>(new S()).Print();
+        new Foo<string>(""hello"").Print();
+        new Foo<string>().Print();
+    }
+}";
+            var verify = CompileAndVerify(source, expectedOutput: @"0
+1
+
+
+0
+0
+hello
+hello");
+
+            verify.VerifyIL("Foo<T>.Print()", @"
+{
+  // Code size       75 (0x4b)
+  .maxstack  2
+  .locals init (T V_0)
+  IL_0000:  ldarg.0
+  IL_0001:  ldflda     ""T Foo<T>.t""
+  IL_0006:  ldloca.s   V_0
+  IL_0008:  initobj    ""T""
+  IL_000e:  ldloc.0
+  IL_000f:  box        ""T""
+  IL_0014:  brtrue.s   IL_002a
+  IL_0016:  ldobj      ""T""
+  IL_001b:  stloc.0
+  IL_001c:  ldloca.s   V_0
+  IL_001e:  ldloc.0
+  IL_001f:  box        ""T""
+  IL_0024:  brtrue.s   IL_002a
+  IL_0026:  pop
+  IL_0027:  ldnull
+  IL_0028:  br.s       IL_0035
+  IL_002a:  constrained. ""T""
+  IL_0030:  callvirt   ""string object.ToString()""
+  IL_0035:  call       ""void System.Console.WriteLine(string)""
+  IL_003a:  ldarg.0
+  IL_003b:  ldfld      ""T Foo<T>.t""
+  IL_0040:  box        ""T""
+  IL_0045:  call       ""void System.Console.WriteLine(object)""
+  IL_004a:  ret
```

Comment:
to be clear for anyone else reading, there is no change in the IL produced here compared to before this PR.